tipg/factory.py

```diff
@@ -444,8 +447,9 @@ def collections(
                 collections_list = [
                     collection
                     for collection in collections_list
-                    if collection.bounds is not None
-                    and s_intersects(bbox_filter, collection.bounds)
+                    if collection.extent is not None
+                    and collection.extent.spatial is not None
+                    and s_intersects(bbox_filter, collection.extent.spatial.bbox)
```

Comment:
the collection's spatial extent is the sum of all the column extent so we can use `collection.bounds` here